src/transaction.rs

```diff
@@ -53,6 +55,7 @@ pub enum Transaction {
         inputs: Vec<Input>,
         outputs: Vec<Output>,
         witnesses: Vec<Witness>,
+        #[derivative(PartialEq = "ignore", Hash = "ignore")]
         metadata: Option<Metadata>,
```

Comment:
We have the same field in the `Create` variant, could you add it there too please?<3